### ISSUE TITLE
Bug: Blank transaction screen

### ISSUE THREAD

**Issue by pyccino** [victorbalssa/abacus#292]
**Describe the bug**
After the initial configuration, even if in the account there are transactions, after clicking the "transactions" tab, the app loads for a quick second and then turns into a full blank screen.

**To Reproduce**
-> Login into the app using the 0.16.0 version of the app
-> Click on the transaction tab

**Expected behavior**
It should show the transaction available for that account on firefly

**Screenshots**
![image](https://github.com/victorbalssa/abacus/assets/76261261/dde771a1-35a9-4e68-9ab0-fff6a5c31819)

**Smartphone (please complete the following information):**
 - Device: IPhone 15
 - OS: IOS 17
 - Version 17.4.1

**Additional context**
My IPhone is set to Italian language (not sure if relevant).
The app loads for a quick second before showing the blank screen

![image](https://github.com/victorbalssa/abacus/assets/76261261/592760c7-c7c5-49db-ba0d-bfe55f7d0479)

There are transactions in the firefly account. But not all "asset accounts" have transactions.

The firefly III installation works with no problems, and transactions can be seen there.

The app uses an Access Token in order to login into firefly III

Other tabs are visible and usable as normal

**Comment by pyccino** [comment]
I've just tried again using the Oauth login and the same bug happened again

Edit:

The bug "fixes" it's self if I chose in the homepage something in the future (such as Q3 2024).
The problem 100% lays if the fact that if there are no transaction in a set period the app "crashes".

After setting Q3 2024 -> transaction I see this
![image](https://github.com/victorbalssa/abacus/assets/76261261/521b9992-36b0-43a5-9c2e-ce0e5c4ccdfb)

(Expected behaviour). If I change the date to anything in May or April (even on specific dates where I do have transactions) it blanks.

But if I set either the Currency or the account to one where I do have transactions it works without bugs
